### DISCUSSION TITLE
How to set the VTextField to required?

### DISCUSSION THREAD

**Discussion by Theodliu** [Kitware/trame#385] [Q&A] [answered]
I'm working with a Vuetify VForm and I've run into a bit of a snag. I have a VTextField that I need to make it mandatory. Despite setting required=True, it doesn't seem to enforce the field as a required input. I'm unsure how to properly implement rules in Python to achieve this. Could someone provide a snippet of Python code that shows how to set up a validation rule for a VTextField within a VForm to make it compulsory?

Here's my code:
`
with vuetify.VForm(
):
        vuetify.VTextField(
        label="defectName",
        v_model=("defectForm.defectName", ),
        required=True,
)
`

Thanks a lot.

**Comment by jourdain** [comment, marked as answer]
Since that `required` property is not listed in Vuetify API it is getting missed by trame, but can dynamically be added by using `__properties=` like described in the API [here](https://trame.readthedocs.io/en/latest/core.widget.html).

```python
    vuetify.VTextField(
        label="defectName",
        v_model=("defectForm",),
        required=True,
        __properties=["required"],
    )
```

[Full example showing the solution](https://github.com/Kitware/trame/blob/master/examples/validation/discussion/385.py)

**Reply by callback00** [reply]
Hello, I would like to set rules for VTextField, but I'm not familiar with its syntax. Could you please add an example of how to use rules in the above code? I would like to achieve the effect of displaying a prompt "Please enter a value" when VTextField is empty.

**Reply by callback00** [reply]
Just like this effect:
![2](https://github.com/Kitware/trame/assets/23352461/38cc7012-bf26-45cd-9898-3054212a2019)

**Reply by jourdain** [reply]
Look at the vuetify properties:
- [V2](https://v2.vuetifyjs.com/en/components/text-fields/)
- [V3](https://vuetifyjs.com/en/components/text-fields/#usage)
